src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/ProjectCapabilityResolver.cs

```diff
@@ -120,20 +123,31 @@ private bool ContainingProjectHasCapability(string capability, string documentFi
             return false;
         }
 
+        var isMatch = false;
         try
         {
-            return vsHierarchy.IsCapabilityMatch(capability);
+            isMatch = vsHierarchy.IsCapabilityMatch(capability);
+
+            if (vsHierarchy.GetProjectFilePath(_jtf) is { } projectFilePath)
+            {
+                _cachedCapabilities = _cachedCapabilities.SetItem((projectFilePath, capability), isMatch);
```

Comment:
How does this cache get invalidated if the project is reloaded -- for example, if the user does a `git pull` and capabilities change?

Reply:
It doesn't. As below, this is not intended for critical uses (and i've updated the comments to reflect that).